postprocessing/comstockpostproc/comstock_to_eia_comparison.py

```diff
@@ -23,21 +24,34 @@ def __init__(self, comstock_list: List[ComStock], eia_list: List[EIA], image_typ
         self.eia_list = eia_list
         self.monthly_data = None
         self.monthly_data_gap = None
+        self.emissions_data = None
+        self.emissions_cols = []
         self.color_map = {}
         self.image_type = image_type
         self.name = name
+        self.column_for_grouping = self.DATASET
+
+        # Get the EIA emission columns
+        for dataset in eia_list:
+            self.emissions_cols = dataset.emissions_cols + [self.DATASET]
 
         # Concatenate the datasets and create a color map
         dfs_to_concat = []
+        emissions_dfs = []
         dataset_names = []
         for dataset in (eia_list + comstock_list):
             if dataset.monthly_data is None:
                 logger.warning(f'No monthly_data was available for {dataset.dataset_name}, not including in EIA comparison.')
                 continue
             if isinstance(dataset, ComStock):
-                dfs_to_concat.append(dataset.monthly_data.to_pandas())
+                # Add the dataset column to ComStock if missing
+                if self.DATASET not in dataset.data.columns:
+                   dataset.add_dataset_column()
+                dataset.data = dataset.data.with_columns(pl.col(self.DATASET).cast(pl.Utf8))
+                emissions_dfs.append(dataset.data.select(self.emissions_cols))
```

Comment:
I've tried this on a couple runs which had the emissions_reporting measure, but keep getting a TypeError in the concat call below at line 65, I think because emissions_cols aren't found in the comstock data.